metrics-core/src/main/java/com/codahale/metrics/ChunkedAssociativeLongArray.java

```diff
@@ -3,15 +3,36 @@
 import static java.lang.System.arraycopy;
 import static java.util.Arrays.binarySearch;
 
+import java.lang.ref.SoftReference;
+import java.util.ArrayDeque;
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.ListIterator;
 
 class ChunkedAssociativeLongArray {
     private static final long[] EMPTY = new long[0];
     private static final int DEFAULT_CHUNK_SIZE = 512;
+    private static final int MAX_CACHE_SIZE = 128;
 
     private final int defaultChunkSize;
+    /*
+    We use this ArrayDeque as cache to store chunks that are expired and removed from main data structure.
```

Comment:
Please use multi-line comments with an asterix on every line. We are trying to follow the Google Java Style Guide. See section about [comments](https://google.github.io/styleguide/javaguide.html#s4.8.6-comments).